### DISCUSSION TITLE
Paste accordion with customizable header

### DISCUSSION THREAD

**Discussion by josamit** [twilio-labs/paste#603] [General] [answered]
Hey team,

I was wondering if the accordion component for paste is ready for GA? We have a requirement where we need the accordion with a customizable header (please refer the screenshot below)
![image](https://user-images.githubusercontent.com/62208722/88368519-efc9f500-cd96-11ea-8d11-b8a4b2c552a2.png)

Do you have any plans to incorporate such a functionality to the accordions? we are using console components at the moment since accordion is not available yet.

**Comment by SiTaggart** [comment, marked as answer]
The accordion for Paste is not yet built, it's coming up pretty soon and we hope to address some of the problems that you are describing with the Console React-Components accordion like not being able to compose different experiences in the headers.

One suggestion that I can offer right now is exploring the Contained Disclosure in Paste. https://paste.twilio.design/components/disclosure#contained-disclosure. It's what we we likely be using as the basis of the Accordion when we do build it.

It may not act entirely like an accordion, but you can string a few of them together and achieve a very similar result. Would you be open to exploring that option?

**Reply by josamit** [reply]
Thanks, I will try Disclosures and then move to accordions once they become available.

**Comment by kate-porter** [comment]
Hey @josamit - just wanted to check in to see if Simon's reply answered your question.

**Reply by josamit** [reply]
Hey Hey @katieporter , my apologies for delayed response. Yes, it seems like Disclosures would work for us at the moment and later when accordions are available,  we would like to switch to them.

**Reply by kate-porter** [reply]
Awesome! Glad to hear it. I'll go ahead and mark this one as answered.

**Comment by elainelylee** [comment]
I see the accordion upcoming in the next couple weeks. What do you see are going to be big differences to be aware of as I'm starting to compose a disclosure while waiting for an accordion. I don't know if it's easier for our team to upgrade to the accordion via console react or the Paste Disclosure.

**Reply by kate-porter** [reply]
Hey @elainelylee - we haven't started work on the accordion yet, so I can't confidently speak to what the differences will be between that and the disclosure. But in general, it will always be easier to transition from one Paste composition/component to another, so I'd recommend going with the disclosure composition for now if it suits your needs, and we can keep you in the loop when the accordion is available.
